cli/src/pcluster/models/cluster.py

```diff
@@ -230,6 +231,12 @@ def _get_cluster_config(self):
                 e, f"Unable to load configuration from bucket '{self.bucket.name}/{self.s3_artifacts_dir}'.\n{e}"
             )
 
+    def _check_bucket_existence(self):
+        try:
+            return self.bucket
```

Comment:
bucket property is cached, and bucket exists check is executed at the first call of this property in the init_s3_bucket, is it necessary to check cached property in all methods?